bin/inx-chronicle/src/api/config.rs

```diff
@@ -0,0 +1,45 @@
+// Copyright 2022 IOTA Stiftung
+// SPDX-License-Identifier: Apache-2.0
+
+use derive_more::From;
+use serde::{Deserialize, Serialize};
+use tower_http::cors::AllowOrigin;
+
+use super::error::ConfigError;
+
+/// API configuration
+#[derive(Clone, PartialEq, Eq, Debug, Serialize, Deserialize)]
+#[serde(default)]
+pub struct ApiConfig {
+    pub port: u16,
+    pub allow_origins: Option<SingleOrMultiple<String>>,
+}
+
+impl Default for ApiConfig {
+    fn default() -> Self {
+        Self {
+            port: 9092,
+            allow_origins: Some(String::from("*").into()),
+        }
+    }
+}
+
+#[derive(Clone, PartialEq, Eq, Debug, Serialize, Deserialize, From)]
+#[serde(untagged)]
+pub enum SingleOrMultiple<T> {
```

Comment:
could you add some docs for this type? I suspect that using `Vec<T>` instead and checking its length is not enough but I'm not sure why